README.md

```diff
@@ -2,4 +2,5 @@ This repository contains a list of tests in the Salt Test Suite that are **not**
 tests is bad and should only be used as a last resort.
 
 The list is formatted using the TOML markup language and skipped tests are grouped by file/test level and then by "type". Missing
-groups do  not have t be listed explicitely.
\ No newline at end of file
+groups do  not have t be listed explicitely.
```

Comment:
Can you please fix the two issues in this line while touching it?
- there are two spaces between `do  not`
- `have t` should probably be `have to`

As an unrelated sidenote, let me add two recommendations when it comes to skipping tests:
- add comments _why_ a specific test gets skipped
- ideally run the skipped tests nevertheless with the expectation that they fails - and bail out if one of them "surprisingly" succeeds. This helps to keep the skip list clean and short.

Reply:
Thanks, the README looks better now :-)

I'd still recommend to have comments for each skipped test (including the initial list, even if this might be difficult). (If you need a BoFH to remove some of the non-commented skiplist entries at random times, tell me ;-)

Also, speaking from experience with the AppArmor unittests, xfail is more than "nice" IMHO, so even if it's difficult to implement, it's worth it. Maybe you can even convince upstream salt to include the xfail handling so that you don't need to carry a patch for conftest.py?

Reply:
The problem with adding comments right now is that I'd need to go through all entries to find out why they are listed. That's work we want to schedule for later: everything will be reviewed and rectified in code (salt or tests) / commented why it has to stay. I know it's important to do it, but if I'd rather get the testing infrastructure in place first and then look at the test failures in detail later.

> Maybe you can even convince upstream salt to include the xfail handling so that you don't need to carry a patch for conftest.py?

In the past the conftest added the tests to xfail. This list only really makes sense as a downstream amendment: tests can be marked as xfail directly in the test code. Tests that upstream sees as unstable are not what we need to avoid (they take care of those), it's the tests that are only failing for us (with our 87 patches). Sadly it does not seem possible to mark a test as xfail via the `pytest` cli command. If that was the case, I'd definitely use that and not `--deselect`. For the current scope of openSUSE/salt-test, this is sadly out of scope. We might reconsider it in the future though.

Reply:
No worries, I understand that checking the failure reason for > 100 tests in the existing list will take some time. That's why I wrote "recommend" and not "have to" 

That only leaves one question: can you define what "later" means? ;-)

For xfail: Agreed, upstream shouldn't skip tests that need xfail (at least not via a list) - but it would be nice if they'd offer an easy way for downstream to set a list of tests to xfail so that xfail'ing a test would be as easy (or easier) as skipping it.

Reply:
Later here means after finishing the remaining ~5 tasks for finishing the integration with (test) infrastructure tools used for testing Uyuni.